### PR TITLE
Task av/j214 store podcast filters in the query functionality

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/common/enums/app/app-route.enum.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 enum AppRoute {
   ROOT = '/',
+  ROOT_$QUERY = '/:params',
```
Comment (what1s1ove) [inline frontend/src/common/enums/app/app-route.enum.ts]:
```suggestion
```

File: frontend/src/components/app/app.tsx [hunk 2]
```diff
@@ -90,7 +90,7 @@ const App: React.FC = () => {
           component={EpisodeLive}
           exact
         />
-        <AuthPublicRouter path={AppRoute.ROOT} component={Homepage} exact />
+        <AuthPublicRouter path={[AppRoute.ROOT, AppRoute.ROOT_$QUERY]} component={Homepage} exact />
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]:
```suggestion
        <AuthPublicRouter path={AppRoute.ROOT} component={Homepage} exact />
```
params should be like this - https://jabber-my.space/?title=moreover

File: frontend/src/common/enums/enums.ts [hunk 3]
```diff
@@ -18,6 +18,7 @@ export {
   PodcastType,
   PodcastSearchPayloadKey,
   PodcastPeriodicity,
+  PodcastLoadFilterKey,
```
Comment (what1s1ove) [inline frontend/src/common/enums/enums.ts]: in the shared folder we have `PodcastSearchPayload`, `PodcastLoadFilterKey`
in this case we should have only one

File: frontend/src/helpers/http/parse-query-string/parse-query-string.helper.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { parse } from 'query-string';
+
+const parseQueryString = (query: string): Record<string, unknown> => parse(query, { arrayFormat: 'bracket' });
```
Comment (what1s1ove) [inline frontend/src/helpers/http/parse-query-string/parse-query-string.helper.ts]:
```suggestion
const parseQueryString = <T =  Record<string, unknown>>(query: string): T => parse(query, { arrayFormat: 'bracket' });
```

File: frontend/src/components/homepage/homepage.tsx [hunk 5]
```diff
@@ -38,48 +42,72 @@ const Homepage: React.FC = () => {
...
+      }
+
+      const isСonsistentLoad = parsedQuery.offset === podcastsFilter.offset + PODCAST_LOAD_LIMIT;
+      setPodcastsFilter(parsedQuery);
+
+      if (isСonsistentLoad) {
+        dispatch(homepageActions.loadMorePodcasts(parsedQuery));
+      } else {
+        dispatch(homepageActions.loadPodcasts(parsedQuery));
+      }
```
Comment (what1s1ove) [inline frontend/src/components/homepage/homepage.tsx]:
```suggestion
        dispatch(isСonsistentLoad ? homepageActions.loadMorePodcasts(parsedQuery) : homepageActions.loadPodcasts(parsedQuery));
```

File: frontend/src/components/homepage/helpers/get-parsed-query/get-parsed-query.helper.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  const { error } = parseQueryValidationSchema.validate(parsedValues);
+
+  if (error) {
+    return null;
+  }
+
+  return {
+    offset: Number(parsedValues.offset),
+    search: parsedValues.search,
+    genres: parsedValues.genres ? parsedValues.genres.map((genreId) => Number(genreId)) : [],
```
Comment (what1s1ove) [inline frontend/src/components/homepage/helpers/get-parsed-query/get-parsed-query.helper.ts]:
```suggestion
    genres: parsedValues.genres?.map((genreId) => Number(genreId)) ?? [],
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/helpers/http/parse-query-string/parse-query-string.helper.ts (hunk 4 above)
Comment (artem-totality) [reply frontend/src/helpers/http/parse-query-string/parse-query-string.helper.ts]: In this case I have error: Type 'ParsedQuery<string>' is not assignable to type 'T'.
  'T' could be instantiated with an arbitrary type which could be unrelated to 'ParsedQuery<string>'.ts(2322)

**Review by yaroslav207 — DISMISSED** [review]

**Comment by Denidzio** [comment]
LGTM 👍🏻

**Review by Denidzio — DISMISSED** [review]
LGTM 👍🏻

File: shared/src/common/types/podcast/podcast-search-payload.type.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import { PodcastSearchPayloadKey } from '~/common/enums/enums';
```
Comment (what1s1ove) [inline shared/src/common/types/podcast/podcast-search-payload.type.ts]: we use this only on the frontend, right? so move it to the frontend part

File: frontend/src/components/homepage/homepage.tsx [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
   useCallback,
   useDispatch,
   useEffect,
   useState,
+  useLocation,
 } from 'hooks/hooks';
 import { homepage as homepageActions } from 'store/actions';
+import { getStringifiedQuery } from 'helpers/helpers';
 import { Search, PopularUsers } from './components/components';
+import { navigation as navigationService } from 'services/services';
```
Comment (what1s1ove) [inline frontend/src/components/homepage/homepage.tsx]:
```suggestion
```
in react component use `useHistory` hook

File: frontend/src/services/navigation/navigation.service.ts [hunk 9]
```diff
@@ -12,7 +12,7 @@ class Navigation {
     this.#history = history;
   }
 
-  public push(path: AppRoute | string): void {
+  public push(path: AppRoute | string | LocationDescriptor<LocationState>): void {
```
Comment (what1s1ove) [inline frontend/src/services/navigation/navigation.service.ts]:
```suggestion
  public push(path: AppRoute | string): void {
```

File: frontend/src/components/homepage/helpers/get-parsed-query/get-parsed-query.helper.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import { PodcastLoadFilter } from 'common/types/types';
+import { parseQueryString } from 'helpers/helpers';
+import { parsedQuery as parsedQueryValidationSchema } from 'validation-schemas/validation-schemas';
+
+const getParsedQuery = (params: string): (PodcastLoadFilter | null) => {
```
Comment (what1s1ove) [inline frontend/src/components/homepage/helpers/get-parsed-query/get-parsed-query.helper.ts]:
```suggestion
const getParsedQuery = (params: string): PodcastLoadFilter | null => {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**